crates/module-system/sov-state/src/nomt/prover_storage.rs

```diff
@@ -82,22 +81,18 @@ where
         self.is_strict_mode = use_strict_mode;
     }
 
-    fn get_version_to_use(&self, version: Option<SlotNumber>) -> Option<SlotNumber> {
+    fn get_version_to_use(&self, version: Option<SlotNumber>) -> Option<Option<SlotNumber>> {
         if self.is_empty() {
             return None;
         }
         let next_version = self.historical_state.get_next_version();
         match version {
-            None => Some(
-                next_version
-                    .checked_sub(1)
-                    .expect("Next version for non empty storage should be above 0"),
-            ),
+            None => Some(None),
```

Comment:
yammie. Could we please add a comment in which cases we get Some(Some(n), and in which `Some(None)` and how it is different from plain `None`.